fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -206,10 +210,10 @@ private CompletableFuture<Void> loadFileSequenceCounter() {
             // Already loaded
             return AsyncUtil.DONE;
         }
-        return context.ensureActive().get(sequenceSubspaceKey).thenAccept(serializedValue -> {
+        CompletableFuture<Long> loadValue = readTransaction(tr -> tr.get(sequenceSubspaceKey).thenApply(FDBDirectory::deserializeFileSequenceCounter));
+        return loadValue.thenAccept(loadedValue -> {
             // Replace the counter value in the counter unless something else has already
             // updated it, in which case we want to keep the cached value
-            long loadedValue = deserializeFileSequenceCounter(serializedValue);
             fileSequenceCounter.compareAndSet(originalValue, loadedValue);
```

Comment:
One of the premises of the way that the `FDBDirectory` was written before was that because of how everything was, the only thing that could update the file sequence counter was either (1) in another transaction or (2) using the same `FDBDirectory` object. And because the file counter (and `fileReferenceCache`) was re-read with every transaction, it would never be stale in a problematic way. However, it seems like that could happen with the multi-transactional directory, if (for example), there's a multi-transactional directory open, and the file sequence counter is updated between transactions. Do we need to rethink that somehow?

Reply:
The all or nothing isolation of changes that include counter updates is lost, of course, now that there are multiple transactions. But within each transaction, I believe the invariants still hold: the counter is initialized as of the start of the transaction and therefore sees updates from earlier in the same directory as well as other transactions in the meantime. So that new references won't be sequential, but I don't know that they need to be.

There is the problem of transaction conflicts after the first commit, which may fail the entire merge process. This will leave some partial writes. But that is the same as if a merging process crashed after some files had be written to the file system. There is orphan junk, but I don't think anything worse than that. Do you?